src/external/SMIOL/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+
+cmake_minimum_required(VERSION 3.10)
+
+find_package(MPI REQUIRED COMPONENTS C Fortran)
+find_package(NetCDF REQUIRED COMPONENTS Fortran C)
+find_package(PnetCDF REQUIRED COMPONENTS Fortran C)
+
+# Specify the library source files
+set(SMIOL_C_SOURCES smiol.c smiol_utils.c)
```

Comment:
`SMIOL_C_SOURCES` and `SMIOLF_SOURCES` don't quite follow the same naming convention. Doesn't matter too much, but it might be nice to align the two. My understanding is `SMIOLF_SOURCES` is meant to be effectively, `<TARGET_NAME>_SOURCES` and follow the Makefile target name.  `SMIOL_SOURCES` for the C sources follows that same convention, though it does *look* more ambiguous than the `_C_` one.

Reply:
Do you mean change `SMIOLF_SOURCES` to `SMIOL_F_SOURCES`?

Reply:
I think that would also work, but in keeping with the `<TARGET_NAME>_SOURCES` I propose `SMIOL_C_SOURCES` => `SMIOL_SOURCES`. If that's too ambiguous of a variable name, then yes exactly what you said instead.

Reply:
I prefer to leave the explicit _C_ naame, esp as c sources are rare.